test/test_profiler.py

```diff
@@ -1147,10 +1147,21 @@ def test_utils_compute_queue_depth(self):
         # so kernel will queued up.
         # But later tensor indexing is overhead bound, and there
         # is sleep to make sure kernel finished before next dispatch.
-        golden_queue_depth_list = [1, 2, 3, 4, 5, 1, 1, 1]
+        golden_queue_depth_list = [
```

Comment:
I'm concerned by the magic numbers in the golden tests; it's very hard to actually verify if they make sense, or diagnose deviations. One thing that would make this easier to follow would be to stringify the expected result and then use expecttest. For example something like:
```
 0  [CPU]
 0  [cudaKernelLaunch]
 1  [CPU]
 1  [cudaKernelLaunch]
 2  [CPU]
 2  [cudaKernelLaunch]
    [GPU]
    [GPU]
    [GPU]
 0  [CPU]
...
```
(Feel free to play with the format.)